tests/generation/test_utils.py

```diff
@@ -1372,6 +1373,26 @@ def test_assisted_decoding_sample(self):
 
             self._check_outputs(output_assisted, input_ids, model.config, use_cache=True)
 
+    def test_prompt_lookup_decoding_stops_at_eos(self):
+        # This test ensures that the prompt lookup generation stops at eos token and does not suggest more tokens
+        # (see https://github.com/huggingface/transformers/pull/31301)
+
+        input_ids = torch.randint(1, 50, (1, 10), device=torch_device)  # generate inputs in range from 1-50
+        input_ids[:, 4] = 0  # inject arbitrarily eos-token-id in input ids so that PLD can copy it
+        pre_eos_token = 51
+        input_ids[:, 3] = pre_eos_token  # set pre-eos to arbitrary id which is for sure not present in inputs
+        input_ids[:, -1] = pre_eos_token  # put pre-eos token in the end for the necessary match to happen
+        eos_token_id = torch.tensor([0], device=torch_device)
+
+        # init cand geenerator with max_matching_ngram_size=1 to match per-token
+        cand_generator = PromptLookupCandidateGenerator(
+            eos_token_id=eos_token_id, num_output_tokens=4, max_matching_ngram_size=1
```

Comment:
Just to make sure I've understood correctly - the value of `num_output_tokens` is independent of the position we injected the eos token id, right (`input_ids[:, 4] = 0`) ?

Reply:
Yes, in any case we should not be proposing new candidates, which means we return the input as is. I can even make a check that we got same values as input ids